woningwaardering/_woningwaardering.py

```diff
@@ -0,0 +1,100 @@
+import warnings
+from datetime import date
+
+from loguru import logger
+
+from woningwaardering.stelsels import utils
+from woningwaardering.stelsels.onzelfstandige_woonruimten.onzelfstandige_woonruimten import (
+    OnzelfstandigeWoonruimten,
+)
+from woningwaardering.stelsels.zelfstandige_woonruimten.zelfstandige_woonruimten import (
+    ZelfstandigeWoonruimten,
+)
+from woningwaardering.vera.bvg.generated import (
+    EenhedenEenheid,
+    WoningwaarderingResultatenWoningwaarderingResultaat,
+)
+from woningwaardering.vera.referentiedata.woningwaarderingstelsel import (
+    Woningwaarderingstelsel,
+)
+
+
+class Woningwaardering:
+    """Initialiseert een Woningwaardering object dat automatisch het juiste stelsel detecteert.
+
+    Parameters:
+        peildatum (date, optional): De peildatum voor de waardering.
+            Standaard is de huidige datum.
+
+    Attributes:
+        peildatum (date): De peildatum voor de waardering.
+        stelsels (dict): Dictionary met alle beschikbare stelsels.
+    """
+
+    def __init__(self, peildatum: date = date.today()) -> None:
+        self.peildatum = peildatum
+
+    def bereken(
+        self,
+        eenheid: EenhedenEenheid,
+    ) -> WoningwaarderingResultatenWoningwaarderingResultaat:
+        """Berekent de woningwaardering voor een eenheid door automatisch het juiste stelsel te detecteren.
+
+        Parameters:
+            eenheid (EenhedenEenheid): De eenheid waarvoor de woningwaardering wordt berekend.
+
+        Returns:
+            WoningwaarderingResultatenWoningwaarderingResultaat: Het resultaat van de woningwaardering.
+
+        Raises:
+            ValueError: Als het type woonruimte niet kan worden bepaald.
+        """
+        if (
+            eenheid.woningwaarderingstelsel is None
+            or eenheid.woningwaarderingstelsel.code is None
+        ):
+            raise ValueError(
+                f"Eenheid {eenheid.id}: woningwaarderingstelsel-attribuut ontbreekt in het inputmodel."
```

Comment:
```suggestion
                f"Eenheid {eenheid.id}: attribuut 'woningwaarderingstelsel' ontbreekt in het inputmodel."
```
zo beter? @tiddoloos

Reply:
haha jep :)